source/agora/common/Deserializer.d

```diff
@@ -235,6 +276,28 @@ public T deserializeFull (T) (
         static assert(0, "Unhandled type: " ~ T.stringof);
 }
 
+/// Options that configure the behavior of the deserializer
+public struct DeserializerOptions
+{
+    /// The bound to apply to a length deserialization (e.g. for arrays)
+    public size_t maxLength = DefaultMaxLength;
+
+    /// Whether or not to use compact notation for unsigned integer
+    public CompactMode compact = CompactMode.Yes;
```

Comment:
How about  `CompactMode.No` as the default? 
`CompactMode.Yes` is only used in special cases.

Reply:
`CompactMode.Yes` is currently the default, hence why I made it the default here.
I am not sure whether it would be good to make it `No` by default. Bear in mind that it only applies to unsigned integers: `ushort`, `uint`, `ulong`. We have a lot of the later (lengths), I doubt we have much of the rest.

In any case, I think changing the default should be in another PR.

Reply:
Yes, if we have to change it, it would be better to do it on another PR.